test/e2e/md_scale_test.go

```diff
@@ -62,16 +62,18 @@ var _ = Describe("When testing MachineDeployment scale out/in", func() {
 
 		By("Creating byohost capacity pool containing 5 hosts")
 		for i := 0; i < byoHostCapacityPool; i++ {
-			byoHostName = fmt.Sprintf("byohost-%s", util.RandomString(6))
-			output, byohostContainerID, err := setupByoDockerHost(ctx, clusterConName, byoHostName, namespace.Name, dockerClient, bootstrapClusterProxy)
-			allbyohostContainerIDs = append(allbyohostContainerIDs, byohostContainerID)
-			Expect(err).NotTo(HaveOccurred())
-
-			// read the log of host agent container in backend, and write it
-			agentLogFile := fmt.Sprintf("/tmp/host-agent-%d.log", i)
-			f := WriteDockerLog(output, agentLogFile)
-			defer f.Close()
-			allAgentLogFiles = append(allAgentLogFiles, agentLogFile)
+			func() {
```

Comment:
Is this a common pattern to define anonymous function inside for loop? I don't understand the difference / improvement from what it was earlier.

Reply:
We can extract out a func too for simplicity, the whole idea is not to miss the deferred error and this being a loop further complicates it

Reply:
i still need to catch the deferred error though

Reply:
Done with catching deferred error, still keeping it anon

Reply:
Could we instead not use `defer` inside `for` loop. The improved version seems a bit hard to read.

Reply:
the file is deferred close so that we can keep writing the docker logs till the function exits, if we close it just after the WriteDocker func call the purpose will be lost as that function does sync to agent log files in go routines AFAIR